packages/next/src/server/resume-data-cache/resume-data-cache.ts

```diff
@@ -34,26 +34,9 @@ export interface MutableResumeDataCache {
  * Creates a new mutable resume data cache. This cache can be mutated and then
  * sealed to create an immutable version of the cache.
  */
-export function createMutableResumeDataCache(): MutableResumeDataCache {
+export function createPrerenderResumeDataCache(): PrerenderResumeDataCache {
```

Comment:
I'd maybe implement in addition to this

```
export function createRenderResumeDataCache(prerenderCache: PrerenderResumeDataCache): RenderResumeDataCache
export function createREnderResumeDataCache(persistedCache: string): RenderResumeDataCache
```

In fact I'd suggest we actually serialize the cache's inner state using `react-server-dom-webpack` rather than `JSON.stringify` so we can just write the object and two inner maps and then reconstruct them on the other side. Then there is no need to have special knowledge of the inernal state when parsing this part of the postponed state, just slice and pass it into `createRenderResumeCache`. This would of course require it to be async but that shouldn't be a problem I think